src/main/scala/scalaz/parsers/transform.scala

```diff
@@ -5,7 +5,18 @@ import scalaz.parsers.implicits.monadKleisliCategory
 import scalaz.tc.ProfunctorClass.DeriveDimap
 import scalaz.tc._
 
-trait TransformClass[=>:[_, _]] extends StrongClass[=>:] with CategoryClass[=>:] {
+trait CategoryClassOps[=>:[_, _]] extends { self: CategoryClass[=>:] =>
+
+  implicit class ArrowOps[B, C](f: B =>: C) {
+    def ∘ [A](g: A =>: B): A =>: C = compose[A, B, C](f, g)
+  }
+
+}
+
+trait TransformClass[=>:[_, _]]
```

Comment:
The simplest way to have `def ∘ [A](g: A =>: B): A =>: C` available inside the trait (and anything extending it) is to simply define it inside the trait. It delegates to `compose` which is also available.

And then the syntax extension is used for anything outside the trait.